File: CleanCode/Net/Aircompany/Airport.cs
using Aircompany.Models;
using Aircompany.Planes;
using System.Collections.Generic;
using System.Linq;

namespace Aircompany
{
    public class Airport
    {
        public List<Plane> planes;

        public Airport(IEnumerable<Plane> _planes)
        {
            planes = _planes.ToList();
        }

        public List<PassengerPlane> GetPassengersPlanes()
        {
            return planes.Where(plane => plane.GetType() == typeof(PassengerPlane)).Cast<PassengerPlane>().ToList();
        }

        public List<MilitaryPlane> GetMilitaryPlanes()
        {
            return planes.Where(plane => plane.GetType() == typeof(MilitaryPlane)).Cast<MilitaryPlane>().ToList();
        }

        public PassengerPlane GetPassengerPlaneWithMaxPassengersCapacity()
        {
            return GetPassengersPlanes().Aggregate((w, x) => w.GetPassengersCapacity() > x.GetPassengersCapacity() ? w : x);             
        }

        public IEnumerable<MilitaryPlane> GetTransportMilitaryPlanes()
        {
            return GetMilitaryPlanes().Where(plane => plane.GetPlaneType() == MilitaryType.Transport);
        }

        public Airport GetAirportWithSortedPlanesByMaxFlightDistance()
        {
            return new Airport(planes.OrderBy(w => w.GetMaxFlightDistance()));
        }

        public Airport GetAirportWithSortedPlanesByMaxSpeed()
        {
            return new Airport(planes.OrderBy(w => w.GetMS()));
        }

        public Airport GetAirportWithSortedPlanesByMaxLoadCapacity()
        {
            return new Airport(planes.OrderBy(w => w.GetMaxLoadCapacity()));
        }


        public IEnumerable<Plane> GetPlanes()
        {
            return planes;
        }

        public override string ToString()
        {
            return "Airport{" + "planes=" + string.Join(", ", planes.Select(x => x.GetModel())) + '}';
        }
    }
}

File: CleanCode/Java/src/main/java/Airport.java
// version: 1.1
// made by Vitali Shulha
// 4-Jan-2019

import plane.ExperimentalPlane;
import model.MilitaryTypes;
import plane.MilitaryPlane;
import plane.PassengerPlane;
import plane.Plane;

import java.util.*;
import java.util.stream.Collectors;

public class Airport {
    private List<? extends Plane> planes;

    public Airport(List<? extends Plane> planes) {
        this.planes = planes;
    }

    public List<PassengerPlane> getPassengerPlanes() {
        return (List<PassengerPlane>) this.planes.stream().filter(plane->plane instanceof PassengerPlane)
                .collect(Collectors.toList());
    }

    public List<MilitaryPlane> getMilitaryPlanes() {
        return (List<MilitaryPlane>) this.planes.stream().filter(plane->plane instanceof MilitaryPlane)
                .collect(Collectors.toList());
    }

    public List<ExperimentalPlane> getExperimentalPlanes() {
        return (List<ExperimentalPlane>) this.planes.stream().filter(plane->plane instanceof ExperimentalPlane)
                .collect(Collectors.toList());
    }

    public PassengerPlane getPassengerPlaneWithMaxPassengersCapacity() {
        return (PassengerPlane) getPassengerPlanes().stream().sorted(Comparator
                .comparingInt(PassengerPlane::getPassengersCapacity)).limit(1);
    }

    public List<MilitaryPlane> getTransportMilitaryPlanes() {
        return (List<MilitaryPlane>) getMilitaryPlanes().stream()
                .filter(plane->plane.getType()==MilitaryTypes.TRANSPORT);
    }

    public List<MilitaryPlane> getBomberMilitaryPlanes() {
        return (List<MilitaryPlane>) getMilitaryPlanes().stream()
                .filter(plane->plane.getType()==MilitaryTypes.BOMBER);

    }

    public List<Plane> sortByMaxDistance() {
        return planes.stream().sorted(Comparator.comparingInt(Plane::getMaxFlightDistance)).collect(Collectors.toList());
    }

    public List<Plane> sortByMaxSpeed() {
        return planes.stream().sorted(Comparator.comparingInt(Plane::getMaxSpeed)).collect(Collectors.toList());
    }

    public List<Plane> sortByMaxLoadCapacity() {
        return planes.stream().sorted(Comparator.comparingInt(Plane::getMaxLoadCapacity)).collect(Collectors.toList());

    }

    @Override
    public String toString() {
        StringBuilder stringBuilder=new StringBuilder();
        stringBuilder.append("Airport{").append("Planes=").append(planes.toString()).append(+'}');
        return stringBuilder.toString();
    }

    public List<? extends Plane> getPlanes() {
        return planes;
    }

    public void setPlanes(List<? extends Plane> planes) {
        this.planes=planes;
    }
}
